src/dotnet/commands/dotnet-build/ProjectGraphNode.cs

```diff
@@ -20,7 +21,16 @@ public ProjectGraphNode(Task<ProjectContext> projectContext, IEnumerable<Project
             IsRoot = isRoot;
         }
 
-        public ProjectContext ProjectContext { get { return _projectContextCreator.GetAwaiter().GetResult(); } }
+        public ProjectContext ProjectContext
+        {
+            get
+            {
+                using (_projectContextCreator.IsCompleted ? null : PerfTrace.Current.CaptureTiming("", "Blocking ProjectContext wait"))
```

Comment:
This doesn't seem like a very good use of a property if it requires trace logging inside of it. Property accesses are supposed to be fast.